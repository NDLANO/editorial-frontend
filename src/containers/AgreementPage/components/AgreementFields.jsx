import React from 'react';
import PropTypes from 'prop-types';
import { injectT } from 'ndla-i18n';
import {
  TextField,
  TextAreaField,
  SelectObjectField,
  DateField,
} from '../../../components/Fields';
import { CommonFieldPropsShape } from '../../../shapes';
import Contributors from '../../../components/Contributors/Contributors';

const AgreementFields = props => {
  const { t, commonFieldProps, licenses } = props;

  return (
    <div>
      <TextField
        label={t('agreementForm.fields.title.label')}
        name="title"
        maxLength={300}
        placeholder={t('agreementForm.fields.title.placeholder')}
        {...commonFieldProps}
      />
      <Contributors
        name="rightsholders"
        label={t('form.rightsholders.label')}
        {...commonFieldProps}
      />
      <Contributors
        name="creators"
        label={t('form.creators.label')}
        {...commonFieldProps}
      />
      <SelectObjectField
        name="license"
        label={t('form.license.label')}
        options={licenses}
        idKey="license"
        labelKey="description"
        {...commonFieldProps}
      />
      <TextAreaField
        label={t('agreementForm.fields.content.label')}
        name="content"
        placeholder={t('agreementForm.fields.content.placeholder')}
        rows="15"
        {...commonFieldProps}
      />
      <DateField
        name="validTo"
        locale="nb"
        {...commonFieldProps}
      />
    </div>
  );
};

AgreementFields.propTypes = {
  commonFieldProps: CommonFieldPropsShape.isRequired,
  classes: PropTypes.func.isRequired,
  bindInput: PropTypes.func.isRequired,
  licenses: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string,
      license: PropTypes.string,
    }),
  ).isRequired,
};

export default injectT(AgreementFields);
