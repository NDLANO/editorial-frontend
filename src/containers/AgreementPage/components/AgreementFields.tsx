import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { spacing } from '@ndla/core';
import { LicenseField, DatePicker, ContributorsField } from '../../FormikForm';
import FormikField from '../../../components/FormikField';

const StyledTwoColumn = styled.div`
  display: flex;
  flex-flow: row;
  margin-top: ${spacing.small};
  & > *:nth-child(1) {
    width: 50%;
  }
  & > *:nth-child(2) {
    margin-left: 0.5rem;
    width: 50%;
  }
`;

const StyledDateField = styled(FormikField)`
  display: inline-block;
  margin-top: 0;
`;
const contributorTypes = ['rightsholders', 'creators'];
const AgreementFields = () => {
  const { t, i18n } = useTranslation();
  const locale = i18n.language;

  return (
    <div>
      <FormikField
        label={t('agreementForm.fields.title.label')}
        name="title"
        maxLength={300}
        placeholder={t('agreementForm.fields.title.placeholder')}
      />
      <ContributorsField contributorTypes={contributorTypes} />
      <FormikField name="license">{({ field }) => <LicenseField {...field} />}</FormikField>
      <FormikField label={t('agreementForm.fields.content.label')} name="content">
        {({ field }) => (
          <textarea
            placeholder={t('agreementForm.fields.content.placeholder')}
            rows="15"
            {...field}
          />
        )}
      </FormikField>
      <StyledTwoColumn>
        <StyledDateField name="validFrom">
          {({ field, form }) => (
            <DatePicker
              label={t('form.validDate.from.label')}
              placeholder={t('form.validDate.from.placeholder')}
              locale={locale}
              {...field}
            />
          )}
        </StyledDateField>
        <StyledDateField name="validTo">
          {({ field, form }) => (
            <DatePicker
              placeholder={t('form.validDate.to.placeholder')}
              label={t('form.validDate.to.label')}
              locale={locale}
              {...field}
            />
          )}
        </StyledDateField>
      </StyledTwoColumn>
    </div>
  );
};

export default AgreementFields;
