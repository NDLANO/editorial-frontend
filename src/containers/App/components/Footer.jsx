/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Footer } from 'ndla-ui';
import SelectLocale from '../../Locale/SelectLocale';

const FooterWrapper = ({ t }) => (
  <Footer>
    <form className="footer_form">
      <label className="footer_label footer--bold" htmlFor="language-select">{t('footer.selectLanguage')}</label>
      <SelectLocale id="language-select" className="footer_language-select" />
    </form>
    <Footer.Ruler />
    <Footer.Text>
      <Footer.Editor title={t('footer.footerEditiorInChief')} name="Øivind Høines" />
      <Footer.Editor title={t('footer.footerManagingEditor')} name="Pål Frønsdal" />
    </Footer.Text>
    <Footer.Text>{t('footer.footerInfo')}</Footer.Text>
  </Footer>
);

FooterWrapper.propTypes = {
  t: PropTypes.func.isRequired,
};

export default FooterWrapper;
