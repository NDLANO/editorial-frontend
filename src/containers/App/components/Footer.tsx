/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React, { FC } from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { spacing } from '@ndla/core';
import { Footer, LanguageSelector, FooterText, EditorName } from '@ndla/ui';
import { Facebook, Twitter, EmailOutline } from '@ndla/icons/common';
import { LocationState } from 'history';
import { withRouter } from 'react-router-dom';
import styled from '@emotion/styled';
//@ts-ignore
import { injectT } from '@ndla/i18n';
import { getLocaleUrls } from '../../../util/localeHelpers';
import { getLocale } from '../../../modules/locale/locale';
import { TranslateType } from '../../../interfaces';

const StyledFooterWrapper = styled.div`
  margin-top: ${spacing.large};
`;

interface Props {
  location: LocationState;
  locale: string;
  t: TranslateType;
  showLocaleSelector?: boolean;
}

export const FooterWrapper: FC<Props> = ({
  location,
  locale,
  showLocaleSelector,
  t,
}) => {
  const languageSelector = showLocaleSelector ? (
    <LanguageSelector
      center
      outline
      alwaysVisible
      options={getLocaleUrls(locale, location)}
      currentLanguage={locale}
    />
  ) : null;

  const links = [
    {
      to: 'https://www.facebook.com/ndla.no',
      text: t('footer.socialMediaLinks.facebook'),
      icon: <Facebook />,
    },
    {
      to: 'https://twitter.com/ndla_no',
      text: t('footer.socialMediaLinks.twitter'),
      icon: <Twitter />,
    },
    {
      to: 'https://om.ndla.no/nyhetsbrev/',
      text: t('footer.socialMediaLinks.newsletter'),
      icon: <EmailOutline />,
    },
  ];

  return (
    <StyledFooterWrapper>
      <Footer lang={locale} links={links} languageSelector={languageSelector}>
        <FooterText>
          <EditorName
            title={t('footer.footerEditiorInChief')}
            name="Sigurd Trageton"
          />
          {t('footer.footerInfo')}
        </FooterText>
      </Footer>
    </StyledFooterWrapper>
  );
};

const mapStateToProps = (state: { locale: string }) => ({
  locale: getLocale(state),
});

export default compose<any>(
  injectT,
  withRouter,
  connect(
    mapStateToProps,
    null,
  ),
)(FooterWrapper);
