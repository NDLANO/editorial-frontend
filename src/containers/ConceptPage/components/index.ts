/*
 * Copyright (c) 2021-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import ConceptArticles from './ConceptArticles';
import ConceptContent from './ConceptContent';
import ConceptCopyright from './ConceptCopyright';
import ConceptMetaData from './ConceptMetaData';

export { ConceptArticles, ConceptContent, ConceptCopyright, ConceptMetaData };
