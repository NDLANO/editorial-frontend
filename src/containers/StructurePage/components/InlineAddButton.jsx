/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'ndla-ui';
import { Done } from 'ndla-icons/editor';
import BEMHelper from 'react-bem-helper';

const classes = new BEMHelper({
  name: 'addButton',
  prefix: 'c-',
});

class InlineAddButton extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      editMode: false,
      inputValue: '',
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  handleClick(e) {
    e.stopPropagation();
    this.props.action(this.state.inputValue);
    this.setState({ editMode: false });
  }

  handleInputChange(e) {
    e.stopPropagation();
    this.setState({ inputValue: e.target.value });
  }

  handleKeyPress(e) {
    if (e.key === 'Escape') {
      this.setState({ editMode: false });
    }
    if (e.key === 'Enter') {
      this.props.action(this.state.inputValue);
      this.setState({ editMode: false });
    }
  }

  render() {
    const { title } = this.props;

    return this.state.editMode ? (
      <div {...classes('editMode')}>
        <input
          type="text"
          data-testid="addSubjectInputField"
          value={this.state.inputValue}
          onChange={this.handleInputChange}
          onKeyDown={this.handleKeyPress}
        />
        <Button stripped onClick={this.handleClick}>
          <Done />
        </Button>
      </div>
    ) : (
      <Button
        stripped
        data-testid="AddSubjectButton"
        onClick={() => {
          this.setState({ editMode: true });
        }}
        {...classes('')}>
        {title}
      </Button>
    );
  }
}

InlineAddButton.propTypes = {
  title: PropTypes.string.isRequired,
  action: PropTypes.func,
};

export default InlineAddButton;
