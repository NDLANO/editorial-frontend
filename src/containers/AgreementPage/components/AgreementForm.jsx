/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree. *
 */

import React, { Component } from 'react';
import { compose } from 'redux';
import { injectT } from '@ndla/i18n';
import { Formik, Form } from 'formik';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Agreement } from '@ndla/icons/editor';
import Field from '../../../components/Field';
import {
  DEFAULT_LICENSE,
  parseCopyrightContributors,
} from '../../../util/formHelper';
import AgreementFields from './AgreementFields';
import { formClasses, FormikActionButton } from '../../FormikForm';
import validateFormik from '../../../components/formikValidationSchema';

const getInitialValues = (agreement = {}) => ({
  id: agreement.id,
  title: agreement.title || '',
  content: agreement.content || '',
  creators: parseCopyrightContributors(agreement, 'creators'),
  processors: parseCopyrightContributors(agreement, 'processors'),
  rightsholders: parseCopyrightContributors(agreement, 'rightsholders'),
  origin:
    agreement.copyright && agreement.copyright.origin
      ? agreement.copyright.origin
      : '',
  license:
    agreement.copyright && agreement.copyright.license
      ? agreement.copyright.license.license
      : DEFAULT_LICENSE.license,
  validFrom:
    agreement.copyright && agreement.copyright.validFrom
      ? agreement.copyright.validFrom
      : undefined,
  validTo:
    agreement.copyright && agreement.copyright.validTo
      ? agreement.copyright.validTo
      : undefined,
});

const rules = {
  title: {
    required: true,
  },
  creators: {
    allObjectFieldsRequired: true,
  },
  processors: {
    allObjectFieldsRequired: true,
  },
  rightsholders: {
    allObjectFieldsRequired: true,
  },
  content: {
    required: true,
  },
  validFrom: {
    required: true,

    dateBefore: true,
    afterKey: 'validTo',
  },
  validTo: {
    required: true,
    dateAfter: true,
    beforeKey: 'validFrom',
  },
};

class AgreementForm extends Component {
  handleSubmit = async (values, actions) => {
    const { licenses, onUpdate } = this.props;
    actions.setSubmitting(true);

    const agreementMetaData = {
      id: values.id,
      title: values.title,
      content: values.content,
      copyright: {
        license: licenses.find(license => license.license === values.license),
        origin: values.origin,
        creators: values.creators,
        processors: values.processors,
        rightsholders: values.rightsholders,
        validFrom: values.validFrom,
        validTo: values.validTo,
      },
    };
    await onUpdate(agreementMetaData);
    actions.setSubmitting(false);
  };

  render() {
    const { t, agreement, licenses, history } = this.props;

    const initVal = getInitialValues(agreement);
    return (
      <Formik
        initialValues={initVal}
        validateOnBlur={false}
        onSubmit={this.handleSubmit}
        enableReinitialize
        validate={values => validateFormik(values, rules, t)}>
        {({ values, isSubmitting }) => (
          <Form {...formClasses('', 'gray-background')}>
            <div {...formClasses('header', 'other')}>
              <div className="u-4/6@desktop">
                <Agreement />
                <span>
                  {values.id
                    ? t('agreementForm.title.update')
                    : t('agreementForm.title.create')}
                </span>
              </div>
            </div>
            <div
              {...formClasses(
                'content',
                '',
                'u-4/6@desktop u-push-1/6@desktop',
              )}>
              <AgreementFields licenses={licenses} />
            </div>
            <Field right>
              <FormikActionButton
                outline
                onClick={history.goBack}
                disabled={isSubmitting}>
                {t('form.abort')}
              </FormikActionButton>
              <FormikActionButton submit>{t('form.save')}</FormikActionButton>
            </Field>
          </Form>
        )}
      </Formik>
    );
  }
}

AgreementForm.propTypes = {
  agreement: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
  }),
  licenses: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string,
      license: PropTypes.string,
    }),
  ).isRequired,
  onUpdate: PropTypes.func.isRequired,
  history: PropTypes.shape({
    goBack: PropTypes.func,
  }).isRequired,
};

export default compose(
  injectT,
  withRouter,
)(AgreementForm);
