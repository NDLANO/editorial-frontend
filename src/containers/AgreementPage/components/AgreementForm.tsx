/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree. *
 */

import { useTranslation } from 'react-i18next';
import { Formik, FormikHelpers } from 'formik';
import { useNavigate } from 'react-router-dom';
import styled from '@emotion/styled';
import { Agreement } from '@ndla/icons/editor';
import { colors } from '@ndla/core';
import { ButtonV2 } from '@ndla/button';
import {
  IAgreement,
  IUpdatedAgreement,
  INewAgreement,
  IAuthor,
} from '@ndla/types-backend/draft-api';
import Field from '../../../components/Field';
import { DEFAULT_LICENSE } from '../../../util/formHelper';
import AgreementFields from './AgreementFields';
import validateFormik from '../../../components/formikValidationSchema';
import { useLicenses } from '../../../modules/draft/draftQueries';
import StyledForm from '../../../components/StyledFormComponents';

interface AgreementFormValues {
  id?: number;
  title: string;
  content: string;
  creators: IAuthor[];
  processors: IAuthor[];
  rightsholders: IAuthor[];
  origin: string;
  license: string;
  validFrom?: string;
  validTo?: string;
}

const getInitialValues = (agreement?: IAgreement): AgreementFormValues => ({
  id: agreement?.id,
  title: agreement?.title || '',
  content: agreement?.content || '',
  creators: agreement?.copyright.creators ?? [],
  processors: agreement?.copyright.processors ?? [],
  rightsholders: agreement?.copyright.rightsholders ?? [],
  origin: agreement?.copyright?.origin || '',
  license: agreement?.copyright?.license?.license || DEFAULT_LICENSE.license,
  validFrom: agreement?.copyright?.validFrom,
  validTo: agreement?.copyright?.validTo,
});

const rules = {
  title: {
    required: true,
  },
  creators: {
    allObjectFieldsRequired: true,
  },
  processors: {
    allObjectFieldsRequired: true,
  },
  rightsholders: {
    allObjectFieldsRequired: true,
  },
  content: {
    required: true,
  },
  validFrom: {
    required: true,

    dateBefore: true,
    afterKey: 'validTo',
  },
  validTo: {
    required: true,
    dateAfter: true,
    beforeKey: 'validFrom',
  },
};

const StyledHeaderContainer = styled.div`
  background-color: ${colors.brand.secondary};
  padding: 0.6rem;
  font-size: 1.1rem;
  color: white;

  svg {
    height: 1.2rem;
    width: 1.2rem;
  }

  svg + span {
    margin-left: 1rem;
  }
`;

interface Props {
  onUpsert: (agreement: IUpdatedAgreement | INewAgreement, id?: number) => Promise<void>;
  agreement?: IAgreement;
}
const AgreementForm = ({ onUpsert, agreement }: Props) => {
  const { t } = useTranslation();
  const { data: licenses } = useLicenses({ placeholderData: [] });
  const navigate = useNavigate();
  const handleSubmit = async (
    values: AgreementFormValues,
    actions: FormikHelpers<AgreementFormValues>,
  ) => {
    actions.setSubmitting(true);

    const agreementMetaData: IUpdatedAgreement = {
      title: values.title,
      content: values.content,
      copyright: {
        license: licenses!.find((license) => license.license === values.license),
        origin: values.origin,
        creators: values.creators,
        processors: values.processors,
        rightsholders: values.rightsholders,
        validFrom: values.validFrom,
        validTo: values.validTo,
      },
    };
    await onUpsert(agreementMetaData, values.id);
    actions.setSubmitting(false);
  };

  const initVal = getInitialValues(agreement);
  return (
    <Formik
      initialValues={initVal}
      validateOnBlur={false}
      onSubmit={handleSubmit}
      enableReinitialize
      validate={(values) => validateFormik(values, rules, t)}
    >
      {({ values, isSubmitting }) => (
        <StyledForm>
          <StyledHeaderContainer>
            <div className="u-4/6@desktop">
              <Agreement />
              <span>
                {values.id ? t('agreementForm.title.update') : t('agreementForm.title.create')}
              </span>
            </div>
          </StyledHeaderContainer>
          <div className="u-4/6@desktop u-push-1/6@desktop">
            <AgreementFields />
          </div>
          <Field right>
            <ButtonV2 variant="outline" disabled={isSubmitting} onClick={() => navigate(-1)}>
              {t('form.abort')}
            </ButtonV2>
            <ButtonV2 type="submit">{t('form.save')}</ButtonV2>
          </Field>
        </StyledForm>
      )}
    </Formik>
  );
};

export default AgreementForm;
