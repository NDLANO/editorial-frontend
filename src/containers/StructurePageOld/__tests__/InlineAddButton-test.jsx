/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { render, fireEvent, cleanup } from '@testing-library/react';
import { InlineAddButton } from '../../../components/InlineAddButton';

afterEach(cleanup);

it('Goes to edit mode, handles input and calls action prop', async () => {
  const actionFunc = jest.fn();
  const { getByTestId, container } = render(<InlineAddButton title="Test" action={actionFunc} />);
  expect(container.firstChild).toMatchSnapshot();

  fireEvent.click(getByTestId('AddSubjectButton'));
  const input = getByTestId('addSubjectInputField');
  fireEvent.change(input, { target: { value: 'Elefant' } });
  expect(container.firstChild).toMatchSnapshot();

  fireEvent.keyDown(input, { key: 'Enter', keyCode: 13, which: 13 });
  expect(container.firstChild).toMatchSnapshot();
  expect(actionFunc).toHaveBeenCalledTimes(1);
});
