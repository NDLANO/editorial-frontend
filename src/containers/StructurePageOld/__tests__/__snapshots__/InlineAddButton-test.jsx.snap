// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goes to edit mode, handles input and calls action prop 1`] = `
.emotion-0 {
  height: 50px;
  margin-left: auto;
  padding: 0 13px;
  white-space: nowrap;
  font-size: 1.1rem;
  background: linear-gradient(
      rgba(226, 226, 226, 0.1),
      #507aa4 50%,
      rgba(226, 226, 226, 0.1) 100%
    );
}

.emotion-0:hover {
  color: white;
}

.emotion-2 {
  display: inline-block;
  color: #ffffff;
  background-color: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  padding: 4px 13px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 700;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  -webkit-transition: background-color none;
  transition: background-color none;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-2:hover,
.emotion-2:focus {
  color: white;
  background-color: #184673;
  border: 2px solid #184673;
}

.emotion-2[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-2:focus {
  box-shadow: 0 0 2px #20588f;
}

.emotion-2:hover,
.emotion-2:active,
.emotion-2:disabled,
.emotion-2:focus {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-2:focus {
  outline-width: medium;
}

<button
  class="emotion-0 emotion-1 emotion-2 emotion-3"
  data-testid="AddSubjectButton"
  type="button"
>
  <svg
    aria-hidden="true"
    class="c-icon"
    data-license="Apache License 2.0"
    data-source="Material Design"
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align: middle;"
    viewBox="0 0 24 24"
    width="1em"
  >
    <g>
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </g>
  </svg>
  Test
</button>
`;

exports[`Goes to edit mode, handles input and calls action prop 2`] = `
.emotion-0 {
  margin: calc(13px / 2);
  width: 200px;
}

<input
  class="emotion-0 emotion-1"
  data-testid="addSubjectInputField"
  type="text"
  value="Elefant"
/>
`;

exports[`Goes to edit mode, handles input and calls action prop 3`] = `
.emotion-0 {
  margin: calc(13px / 2);
  width: 200px;
}

<input
  class="emotion-0 emotion-1"
  data-testid="addSubjectInputField"
  type="text"
  value="Elefant"
/>
`;
